sema/ro/creator/roblueprint.py

```
@@ -36,45 +39,67 @@ def __init__(
+            self.body["**/*"] = {}
+        self._expand_body()
         assert (
             not self.nested_datasets
         ), "nested datasets are not yet supported"  # TODO support nested datasets # noqa: E501
 
-    def _expand_glob(self):
-        assert self.glob_root, "glob_root must be set to expand glob patterns"
-        self.body = {}
-        for glob_pattern, property in self.implicit_body.items():
-            paths = getMatchingGlobPaths(
+    @cache
```

Ruff (0.16.2)

[warning] Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)